### PR TITLE
Create not eligible screen and show when appropriate

### PR DESCRIPTION
**Pull request by SVarallo**

### CODE REVIEW

File: src/controllers/confirmation.controller.ts [hunk 1]
```diff
@@ -70,12 +72,21 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
   if (isStillRequired) {
     const filingDueOn = apiResponseData.filing_due_on;
 
+    if (apiResponseStatus === 400) {
+      const cannotUseReason: string = eligibilityReasonCode[apiResponseData.reason_code];
```
Comment (ICoxCH) [inline src/controllers/confirmation.controller.ts:76]: If somehow apiResponseData.reason_code is null. What happens here (Just thinking if the api returns a 400 for some reason other than the eligibility check fails)

**Comment by mwestacott** [comment]
I can't remember - should/do changes to the package-lock.json file need to be committed?

File: src/controllers/confirmation.controller.ts (hunk 1 above)
Comment (mwestacott) [reply src/controllers/confirmation.controller.ts:76]: Spec says that for a 400 response reason_code should always be set, so if it isn't we should probably just show the 'Sorry, service failed' screen I guess?

File: test/controllers/confirmation.controller.spec.unit.ts [hunk 2]
```diff
@@ -163,6 +164,154 @@ describe("Company no longer required confirmation screen tests", () => {
...
+    loadCompanyAuthenticatedSession(mockCacheService, COMPANY_NUMBER, EMAIL);
+    const dummyProfile = getDummyCompanyProfile(true, true);
+    mockPTFSession.mockImplementationOnce(() => dummyProfile);
+    mockPTFSession.mockImplementationOnce(() => true);
+    mockActiveFeature.mockImplementationOnce(() => true);
+
+    mockCallProcessorApi.prototype.constructor.mockImplementation(() => Promise.resolve((
+      {
+        data: {
+          filing_due_on: "null",
```
Comment (mwestacott) [inline test/controllers/confirmation.controller.spec.unit.ts:183]: According to the spec, filing_due_on field isn't even present in the JSON for a 400 response. So I'm interested in what the API does actually return? Did/could you try removing the "null" value here (and does test still pass)?

File: src/model/eligibilityReasonCode.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export enum eligibilityReasonCode {
+  COMPANY_IS_ELIGIBLE = "company is eligible",
```
Comment (mwestacott) [inline src/model/eligibilityReasonCode.ts:2]: So these values don't seem to match the current return codes from the API - @mattch1 did merge a PR recently which changed these. So I think they need to correspond to what's in the API **_EligibilityReturnCodeToWeb_** Java enum.

Comment (SVarallo) [reply src/model/eligibilityReasonCode.ts:2]: Yep, I'll change the values to match those in the api. The strings okay to stay as they are?

File: test/controllers/confirmation.controller.spec.unit.ts (hunk 2 above)
Comment (SVarallo) [reply test/controllers/confirmation.controller.spec.unit.ts:183]: They pass without the filing_due_on field. I'll remove them 👍

**Comment by SVarallo** [comment]
> I can't remember - should/do changes to the package-lock.json file need to be committed?

Yes, changes to the package-lock do need to be committed 👍

File: src/model/eligibilityReasonCode.ts (hunk 3 above)
Comment (mwestacott) [reply src/model/eligibilityReasonCode.ts:2]: 🆗 Yes, string values are fine, assuming they're still sensible for the values in EligibilityReturnCodeToWeb.

**Pull request merged**